File: 3455-minimum-length-of-string-after-operations/3455-minimum-length-of-string-after-operations.cpp
class Solution {
public:
    int minimumLength(string s) {
        int n = s.size();
        vector<int> charFreq(26,0);

        for(char x : s){
            charFreq[x-'a']++;
        }

        int ans = 0;
        for(int freq : charFreq){
            if(freq == 0){
                continue;
            } else if(freq % 2 == 0){
                ans += 2;
            } else {
                ans += 1;
            }
        }

        return ans;
    }
};

File: 3455-minimum-length-of-string-after-operations/3455-minimum-length-of-string-after-operations.java
class Solution {
    public int minimumLength(String s) {
        int n = s.length();
        int[] charFreq = new int[26];

        for(int i = 0; i < n; i++){
            charFreq[s.charAt(i)-'a']++;
        }


        int ans = 0;
        for(int frequency : charFreq){
            if(frequency == 0) continue;

            if(frequency % 2 == 0){
                ans += 2;
            } else {
                ans += 1;
            }
        }

        return ans;

    }
}

// abaacbcbb
/*

a = 3
b = 4
c = 2



 */